File: Project/C/project/product/test/test/src/service_test/service/base/base_case.cpp
/*
 * Copyright (c) 2022 Renwei
 *
 * This is a free software; you can redistribute it and/or modify
 * it under the terms of the MIT license. See LICENSE for details.
 */
#ifdef __DAVE_PRODUCT_TEST__
#include <iostream>
using namespace std;
#include "gtest.h"
#include "dave_base.h"
#include "test_log.h"

class BaseCase:public testing::Test {
	protected:
		virtual void SetUp() {
		}
		virtual void TearDown() {
		}
};

TEST_F(BaseCase, base_case)
{
	TESTLOG("");

    EXPECT_EQ(4, 3);
}

// =====================================================================

dave_bool
base_case(s8 *input_dir, s8 *output_dir, s8 *gid, s8 *service, ThreadId id)
{
	return dave_true;
}

#endif

File: Project/Go/project/product/test/test/test_service/service/base_case.go
package service

/*
 * Copyright (c) 2022 Renwei
 *
 * This is a free software; you can redistribute it and/or modify
 * it under the terms of the MIT license. See LICENSE for details.
 */

import (
	"dave/product/test/refittesting"
	"dave/public/auto"
	"dave/public/base"
	"unsafe"
)

// =====================================================================

func Base_case(t *refittesting.T) {
	req := auto.RPCDebugReq{}
	req.S16_debug = 12345
	req.Mbuf_debug = nil

	pRsp := (*auto.RPCDebugRsp)(base.Name_go("BASE", auto.MSGID_RPC_DEBUG_REQ, int(unsafe.Sizeof(req)), unsafe.Pointer(&req), auto.MSGID_RPC_DEBUG_RSP))
	if pRsp != nil {
		if pRsp.S16_debug == req.S16_debug {
			t.Log("RPCDebugReq success")
		} else {
			t.Error("RPCDebugReq failed")
		}
	} else {
		t.Error("get empty rsp!")
	}
}
